django/db/models/sql/query.py

```diff
@@ -1704,12 +1704,12 @@ def add_filtered_relation(self, filtered_relation, alias):
                             "relations outside the %r (got %r)."
                             % (filtered_relation.relation_name, lookup)
                         )
-                else:
-                    raise ValueError(
-                        "FilteredRelation's condition doesn't support nested "
-                        "relations deeper than the relation_name (got %r for "
-                        "%r)." % (lookup, filtered_relation.relation_name)
-                    )
+            if len(lookup_field_parts[:-1]) > len(relation_field_parts):
```

Comment:
```suggestion
            if len(lookup_field_parts) > len(relation_field_parts) + 1:
```
I think I prefer this than `lookup_field_parts[:-1]`. I'm also tempted to put this above the loop but I don't think it makes much difference. Let me know what you think :+1:

Reply:
Agree the +1 is better.

Re: moving the + 1 above the loop, there are a lot of instance variables to grok in this block, so I kind of like leaving the +1 where it is. I was torn on whether to include a comment.